src/file_pipe_log/pipe_builder.rs

```diff
@@ -315,7 +318,18 @@ impl<F: FileSystem> DualPipesBuilder<F> {
             LogQueue::Rewrite => &self.rewrite_files,
         };
         let first_seq = files.first().map(|f| f.seq).unwrap_or(0);
-        let files: VecDeque<Arc<F::Handle>> = files.iter().map(|f| f.handle.clone()).collect();
+        let files: VecDeque<FileHandler<F>> = files
+            .iter()
+            .map(|f| {
+                // As the version of each log file could not be captured from outsize, we
+                // need to do the `[build_file_header]` for parsing the `[Version]` at here.
+                let file_header = build_file_header(self.file_system.as_ref(), f.handle.clone());
```

Comment:
Refer to previous comment, no need to decode again.

Reply:
Done